### PR TITLE
Portable Frozen Enemies

### PR DESCRIPTION
**Pull request by weluvgoatz**
A few weeks ago, I suggested that frozen enemies are carryable. And this pull request makes that happen!  I have copied some of the frozen bomb's code to the 13 enemies. BUT, I have no way of testing these (I'm a bit busy) so can you please test these enemies out?

What is supposed to happen: These enemies can be picked up when they're frozen

Thanks!

(Also, for some reason, the forest level commit I made showed up and won't go away. Why?)

### CODE REVIEW

**Comment by weluvgoatz** [comment]
I didn't make some of those changes. I downloaded the files from other pull
requests.  Also, the forest levels wont go away. I tried. I will remove the
changes that I made.

**Comment by maxteufel** [comment]
Please do not close a pull request without reason. It's possible to fix it, and that should be preferred to closing and making an new one.

**Comment by maxteufel** [comment]
@weluvgoatz I fixed the essential problems with this now. I would recommend you (and anyone who wishes to contribute to SuperTux, really) to learn more about Git, for example using their great documentation at <https://git-scm.com/doc> (the "Pro Git" book that you can read there for free seems a good resource). This helps making good pull requests.

**Comment by weluvgoatz** [comment]
thanks a ton. I will learn more about git before making more pull requests.

**Comment by christ2go** [comment]
This still needs/would profit from new sprites which @Rusty-Box wanted to draw. Any progress on this?

**Comment by tobbi** [comment]
Potentially, there is one block in https://github.com/SuperTux/supertux/issues/683. We could reduce the original leaf from it and extract the block and this way easily adapt it to the other enemies.

**Comment by weluvgoatz** [comment]
I think the enemies shouldn't shatter when placed, like @Rusty-Box suggested in #683. They should be placd so Tux can stand or stack them. I can also make sprites, if needed.

File: src/badguy/walkingleaf.cpp [hunk 1]
```diff
@@ -26,10 +26,19 @@ WalkingLeaf::WalkingLeaf(const ReaderMapping& reader) :
...
+{
+  walk_speed = 60;
+  max_drop_height = 16;
+}
+
 bool
 WalkingLeaf::collision_squished(GameObject& object)
 {
   sprite->set_action(dir == LEFT ? "squished-left" : "squished-right");
+  // Spawn death particles
```
[the remark this answers is not in the document]
Comment (weluvgoatz) [reply src/badguy/walkingleaf.cpp]: Didn't do that.

File: src/badguy/walkingleaf.cpp [hunk 2]
```diff
@@ -26,10 +26,19 @@ WalkingLeaf::WalkingLeaf(const ReaderMapping& reader) :
   max_drop_height = 16;
 }
 
+WalkingLeaf::WalkingLeaf(const Vector& pos, Direction d)
```
[the remark this answers is not in the document]
Comment (weluvgoatz) [reply src/badguy/walkingleaf.cpp]: didn't do that either.

**Comment by weluvgoatz** [comment]
sorry for wasting your time on this. i had no idea what i was doing. i am not any sort of coder and don't plan on it. please dont reopen this, i am done with it.

**Comment by weluvgoatz** [comment]
Yikes

**Pull request closed without merging**